### PR TITLE
Simplify and cleanup template app

### PR DESCRIPTION
**Pull request by Supereg**
# Simplify and cleanup template app

## :recycle: Current situation & Problem
This PR tries to do some housekeeping for the template app. Refer to the release notes for a list of changes.

## :gear: Release Notes 
* Updated dependencies.
* Removed Codable-based RawRepresentable conformance for array to avoid anti-patterns with `@AppStorage`.
* Automatically pass `PreferLogStreaming=YES` to the environment to silence warning.
* Remove `setupSimulators`, saving some seconds in the CI.
* Remove MockWebService.

## :books: Documentation
Removed some outdated screenshots related to the MockWebService integration.

## :white_check_mark: Testing
Removed outdated tests.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for all the fixes and improvements; looks great!

File: README.md [hunk 1]
```diff
@@ -37,9 +37,9 @@ The following screenshots show a wide variety of features based on Spezi Modules
...
-|:--:|:--:|:--:|
-|Contact Information|License Information|Mock Web Service Requests|
+|![A screen displaying the Contact UI.](TemplateApplication/Supporting%20Files/TemplateApplication.docc/Resources/Context/Contacts.png#gh-light-mode-only) ![A screen displaying the Contact UI.](TemplateApplication/Supporting%20Files/TemplateApplication.docc/Resources/Context/Contacts~dark.png#gh-dark-mode-only)|![License information to list all used Swift Packages](TemplateApplication/Supporting%20Files/TemplateApplication.docc/Resources/Context/License.png#gh-light-mode-only) ![License information to list all used Swift Packages](TemplateApplication/Supporting%20Files/TemplateApplication.docc/Resources/Context/License~dark.png#gh-dark-mode-only)|
```
Comment (PSchmiedmayer) [inline README.md:40]: One minor thing: Could we add a third screenshot here? Looks a bit strange in the README with only two screenshots here and them scaling to be larger.

Comment (Supereg) [reply README.md]: I added the Account Overview into that Row and updated the order of the Schedule images a bit 👍

**Comment by Supereg** [comment]
I adjusted some of the timeout values in the Fastfile as they were continuously hit by the build jobs. Specifically for CodeQL this should save like 5 minutes (which isn't a lot for the whopping 30-40 minutes build time). Do you think it is possible to offload compilation to our custom runners (e.g., upload derived data as a build artifact and just download it for the CodeQL job). Or does the CodeQL swift extractor inject itself into the build process? Can't quite find any good resources for how exactly the swift support is achieved.

**Pull request merged**